src/qibolab/pulses.py

```diff
@@ -1122,44 +1094,43 @@ def serial(self) -> str:
     def id(self) -> int:
         return id(self)
 
-    @property
-    def envelope_waveform_i(self) -> Waveform:
+    def envelope_waveform_i(self, sampling_rate=SAMPLING_RATE) -> Waveform:
         """The envelope waveform of the i component of the pulse."""
 
-        return self._shape.envelope_waveform_i
+        return self._shape.envelope_waveform_i(sampling_rate)
 
-    @property
-    def envelope_waveform_q(self) -> Waveform:
+    def envelope_waveform_q(self, sampling_rate=SAMPLING_RATE) -> Waveform:
         """The envelope waveform of the q component of the pulse."""
 
-        return self._shape.envelope_waveform_q
+        return self._shape.envelope_waveform_q(sampling_rate)
 
-    @property
-    def envelope_waveforms(self):  #  -> tuple[Waveform, Waveform]:
+    def envelope_waveforms(
+        self, sampling_rate=SAMPLING_RATE
+    ):  #  -> tuple[Waveform, Waveform]:
         """A tuple with the i and q envelope waveforms of the pulse."""
 
-        return (self._shape.envelope_waveform_i, self._shape.envelope_waveform_q)
+        return (
+            self._shape.envelope_waveform_i(sampling_rate),
+            self._shape.envelope_waveform_q(sampling_rate),
+        )
 
-    @property
-    def modulated_waveform_i(self) -> Waveform:
+    def modulated_waveform_i(self, sampling_rate=SAMPLING_RATE) -> Waveform:
         """The waveform of the i component of the pulse, modulated with its
         frequency."""
 
-        return self._shape.modulated_waveform_i
+        return self._shape.modulated_waveform_i(sampling_rate)
 
-    @property
-    def modulated_waveform_q(self) -> Waveform:
+    def modulated_waveform_q(self, sampling_rate=SAMPLING_RATE) -> Waveform:
         """The waveform of the q component of the pulse, modulated with its
         frequency."""
 
         return self._shape.modulated_waveform_q
```

Comment:
Shouldn't also this include the sampling rate?
```suggestion
        return self._shape.modulated_waveform_q(sampling_rate)
```

Reply:
That's correct: the problem is that these methods are just shortcuts to the underlying `PulseShape` ones (and apparently they sounded so redundant that they were not even tested, since in the tests there is the explicit call to `pulse.shape.modulate_waveform_q()`).